source/common/network/utility.cc

```diff
@@ -598,13 +598,13 @@ void passPayloadToProcessor(uint64_t bytes_read, Buffer::InstancePtr buffer,
                                      std::move(buffer), receive_time);
 }
 
-Api::IoCallUint64Result Utility::readFromSocket(IoHandle& handle,
-                                                const Address::Instance& local_address,
-                                                UdpPacketProcessor& udp_packet_processor,
-                                                MonotonicTime receive_time, bool use_gro,
-                                                uint32_t* packets_dropped) {
-
-  if (use_gro) {
+Api::IoCallUint64Result
+Utility::readFromSocket(IoHandle& handle, const Address::Instance& local_address,
+                        UdpPacketProcessor& udp_packet_processor, MonotonicTime receive_time,
+                        UdpRecvMsgMethod recv_msg_method, uint32_t* packets_dropped) {
+  if (recv_msg_method == UdpRecvMsgMethod::RecvMsgWithGro) {
```

Comment:
I thought server side we wanted GRO (if enabled), then mmsg (if supported) then msg.  Doesn't this only allow one at a time now?
/wait-any

Reply:
We decide which method to use in https://github.com/envoyproxy/envoy/pull/32628/files#diff-c9702469f313c70572e261f4af73687873ab8247264eb47fd05300098067c5abR727-R732.

readFromSocket is only called from readPacketsFromSocket (other than in tests)